File: spring-web-java/src/main/java/spring/web/java/interfaces/dto/response/OrderItemResponse.java
package spring.web.java.interfaces.dto.response;

import spring.web.java.domain.model.entity.OrderItem;

public record OrderItemResponse(
	String itemName,
	int orderPrice,
	int count
) {
	public OrderItemResponse(OrderItem orderItem) {
		this(
			orderItem.getItem().getName(),
			orderItem.getOrderPrice(),
			orderItem.getCount()
		);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/presentation/dto/response/OrderItemResponse.kt
package spring.web.kotlin.presentation.dto.response

import spring.web.kotlin.domain.model.entity.OrderItem

data class OrderItemResponse(
    val itemName: String,
    val orderPrice: Int,
    val count: Int,
) {
    constructor(orderItem: OrderItem) : this(
        itemName = orderItem.item.name,
        orderPrice = orderItem.orderPrice,
        count = orderItem.count
    )
}

